app/layout/CmsLayout.tsx

```diff
@@ -0,0 +1,191 @@
+					color={color}>
+					{name}
+				</Text>
+			</Flex>
+		</Link>
+	);
+}
+
+type CurrentPage = "pages" | "users" | "home" | "files";
+
+interface Props {
+	currentPage: CurrentPage;
+}
+
+const LOGOUT = gql`
+	mutation Logout {
+		logout {
+			id
+		}
+	}
+`;
+
+const CURRENT_USER = gql`
+	query User {
+		currentUser {
+			firstname
+			lastname
+		}
+	}
+`;
+
+interface CurrentUser {
+	currentUser: Pick<User, "firstname" | "lastname">;
+}
+
+function CmsLayout({ children, currentPage }: PropsWithChildren<Props>) {
+	const [logout, { loading: logoutLoading, data: logoutData }] =
+		useMutation<{ user: IronSessionUser }>(LOGOUT);
+	const { loading: userLoading, data: userData } =
+		useQuery<CurrentUser>(CURRENT_USER);
+
+	const router = useRouter();
+
+	const { colorMode, toggleColorMode } = useColorMode();
+
+	if (logoutData) {
+		if (typeof window !== "undefined") {
+			void router.push("/login");
+		}
+	}
+
+	return (
+		<Flex w="100vw" h="100vh">
+			<Flex h="100%" w="13%" minW={32} direction="column">
+				<Flex w="100%" bg="brand.500" h={20} pl={5} alignItems="center">
+					<Skeleton isLoaded={!userLoading}>
+						{userData && (
+							<Heading color="white" fontSize={20}>
+								{`${userData.currentUser.firstname} `}
+								{userData.currentUser.lastname}
+							</Heading>
+						)}
+					</Skeleton>
+				</Flex>
+				<Flex
+					flexGrow={1}
+					bg={useColorModeValue("gray.300", "gray.700")}
+					py={5}
+					px={2}
+					direction="column"
+					justifyContent="space-between">
+					<Stack direction="column">
+						<NavElement
+							icon={FiHome}
+							name="Home"
+							href="/cms"
+							selected={currentPage === "home"}
+						/>
+						<NavElement
+							icon={FiFileText}
+							name="Pages"
+							href="/cms/pages"
+							selected={currentPage === "pages"}
+						/>
+						<NavElement
+							icon={FiUsers}
+							name="Users"
+							href="/cms/users"
+							selected={currentPage === "users"}
+						/>
+						<NavElement
+							icon={FiImage}
+							name="Files"
+							href="/cms/files"
+							selected={currentPage === "files"}
+						/>
+					</Stack>
+					<Stack>
+						<Flex
+							mx={2}
+							alignItems="center"
+							justifyContent="space-between">
+							<Button
+								w={10}
+								h={10}
+								onClick={toggleColorMode}
+								variant="outline">
+								{colorMode === "dark" ? (
+									<SunIcon />
+								) : (
+									<MoonIcon />
+								)}
+							</Button>
+							<Button
+								onClick={() => {
+									void logout({});
+								}}
+								isLoading={logoutLoading}
+								variant="outline"
+								colorScheme="red">
+								Logout
+							</Button>
+						</Flex>
+					</Stack>
+				</Flex>
+			</Flex>
+			<Box flexGrow={1}>{children}</Box>
+		</Flex>
+	);
+}
+
+CmsLayout.getLayout = (currentPage: CurrentPage) => {
+	return function CmsLayoutWrapper(page: ReactElement) {
+		return <CmsLayout currentPage={currentPage}>{page}</CmsLayout>;
+	};
+};
```

Comment:
Giving CMS pages the ability to modify their own Next.js layout is going to be tricky, I would stick to leaving everything other than content to the developers